examples/type-conversion/type_conversion.bal

```diff
@@ -1,71 +1,106 @@
 import ballerina/io;
 
 type Person record {
-    string name = "";
-    int age = 0;
+    string name;
+    int age;
 };
 
 type Employee record {
-    string name = "";
-    int age = 0;
-    int empNo = 0;
+    string name;
+    int age;
+    int empNo;
 };
 
-function convertType(Employee emp) returns () {
-    // The `convert()` creates a new value and changes its type without editing provided value's inherent type.
-    Person|error empPerson = Person.convert(emp);
-    io:println("empPerson name: ",
-                (empPerson is Person) ? empPerson["name"] : empPerson.reason());
+// Function to attempt converting an `anydata` record `Employee`, to an `anydata` record `Person`.
+function convertEmployeeToPerson(Employee emp) {
+    // Attempt creating a new value of type `Person` from the `Employee` typed `emp` value, without changing `emp`'s
+    // inherent type.
+    Person|error res = Person.convert(emp);
+    if (res is Person) {
+        // If the conversion is successful, print the `name` field.
+        io:println("Employee to Person, name: ", res["name"]);
+    } else {
+        io:println("error occurred on conversion");
+    }
 }
 
-function convertSimpleTypes() {
+// Function to attempt converting an `any` constrained map, to an `anydata` record `Person`.
+// The conversion would return an error if non-`anydata` or an incompatible value is found.
+function convertAnyMapToPerson(map<any> m) {
+    // Attempt creating a new value of type `Person` from the `map<any>` typed `m` value, without changing `m`'s
+    // inherent type.
+    Person|error res = Person.convert(m);
+    if (res is Person) {
+        // If the conversion is successful, print the `name` field.
+        io:println("map<any> to Person, name: ", res["name"]);
+    } else {
+        io:println("error occurred on conversion");
+    }
+}
 
-    // The `convert()` can be used to explicity convert simple values as below.
+// Function to convert simple basic types using the `.convert()` built-in method.
+function convertSimpleBasicTypes() {
     string s1 = "45";
     string s2 = "abc";
     string s3 = "true";
     float f = 10.2;
     any a = 3.14;
-    
-    int|error intVal1 = int.convert(s1);
-
-    int|error intVal2 = int.convert(s2);
-
-    int intVal3 = int.convert(f);
 
-    boolean|error b = boolean.convert(s3);
-
-    float|error af = float.convert(a);
-
-    if (intVal1 is int) {
-        io:println("Int value 1 : " + intVal1);
+    // `string` to `int` conversion is unsafe, since the `string` may not be convertible to `int`.
```

Comment:
the `string` or the `string` value? referring to the fact that the word string is formatted as code (i.e., the keyword `string`)